content/blog/2019-12-18-polynomial/index.md

```diff
@@ -0,0 +1,155 @@
+work and it is also not a fair comparison among different systems if some
+researchers just don't have the resource to invest a lot of time on crafting
+those highly optimized assembly code. Here, we hope to start addressing this
+problem by creating an code generating system where its goal is to create high
+performance polynomial multiplication code commonly used in a lattice-based
+post-quantum crpyto systems.
+
+### High Performance Code Generation
+
+Generating high performance code to run on different hardware have been studied
+for decades. Signal processing community have been studied how to generate high
+performance FFT implementation, like [FFTW](http://www.fftw.org/) and
+[Spiral](https://www.spiral.net/index.html). In recent years, Image
+processing and machine learning communities also propose several framework to
+generate high performance code, such as on the
+[Halide](https://github.com/halide/Halide), [Loopy](https://github.com/inducer/loopy) and
+[TVM](https://github.com/apache/incubator-tvm). 
+
+## Method 
+
+### Schoolbook Multiplication
+
+This is the straight forward basic method.
+
+### Karatsuba
+We break the polynomial $f(x)$ and $g(x)$ to high and low two part.
+
+
+  $$
+  f(x) = a_1(x) t + a_0(x)
+  $$
+  $$
+  g(x) = b_1(x) t + b_0(x)
+  $$
+  , where $t = x^{\frac{n}{2}}$. 
+  Then the multiplication can be calculated as follows:
+
+  $$
+  f(x)g(x) = c_2(x) t^2 + c_1 t + c_0(x)
+  $$
+
+  where 
+  $$ c_2(x) = a_1(x)b_1(x) $$
+  $$ c_0(x) = a_0(x)b_0(x) $$
+
+  $$
+  c_1(x) = (a_1(x) + a_0(x)) (b_1(x) +b_0(x)) -c_0(x)-c_2(x)
+  $$
+
+  It uses 3 multiplications while the schoolbook method would use 4.
+  The idea can be further generalize to k-way split, and it is called Toom-k.
+
+### Toom-k
+
+Lets take Toom-3 as an example. First we divide the polynomial $f(x)$ and $g(x)$ into 3 parts.
+
+$$
+f(x) = a_2(x) t^2 + a_1(x) t + a_0(x)
+$$
+
+$$
+g(x) = b_2(x) t^2 + b_1(x) t + b_0(x)
+$$
+
+where $t = x^{\frac{n}{3}}$. 
+
+
+Now, we substitute t with values: $0$, $1$, $-1$, $2$, and $\inf$.
+and we let $f(x)g(x) = c_4(x)t^4 + c_3(x)t^3 + c_2(x) t^2 + c_1(x) t + c_0(x)$
+We get the following table:
+
+|  $t=$    |                                                      |                          |
+|---------|------------------------------------------------------|------------------------------------------------|
+|$ 0 $  |$a_0(x)b_0(x)$                                        |$=c_0(x)$                                        |
+|$ 1 $  |$(a_2(x)+a_1(x)+a_0(x)) (b_2(x)+b_1(x)+b_0(x))$       |$=c_4(x) +c_3(x)+c_2(x)+c_1(x)+c_0(x)  $         |
+|$ -1$  |$(a_2(x)-a_1(x)+a_0(x)) (b_2(x)-b_1(x)+b_0(x))$       |$=c_4(x) -c_3(x)+c_2(x) -c_1(x)+c_0(x)  $        |
+|$ 2 $  |$(4a_2(x)+2a_1(x)+a_0(x)) (4b_2(x)+2b_1(x)+b_0(x))$   |$=16c_4(x) +8c_3(x)+4c_2(x)+2c_1(x)+c_0(x)  $    |
+|$\inf$ |$a_2(x)b_2(x)$                                        |$=c_4(x)$                                        |
```

Comment:
I don't really understand what "t" is, or what the rows in the table actually mean? Are these different regimes?

Reply:
Remove the confusing derivation as this can be considered detail not related to the generation technique.